File: src/main/java/com/andrew121410/mc/ccminecraftbot/packets/handle/OnServerChunkDataPacket.java
package com.andrew121410.mc.ccminecraftbot.packets.handle;

import com.andrew121410.mc.ccminecraftbot.CCBotMinecraft;
import com.andrew121410.mc.ccminecraftbot.packets.PacketHandler;
import com.github.steveice10.mc.protocol.packet.ingame.server.world.ServerChunkDataPacket;
import lombok.AllArgsConstructor;

public class OnServerChunkDataPacket extends PacketHandler<ServerChunkDataPacket> {

    @Override
    public void handle(ServerChunkDataPacket packet, CCBotMinecraft ccBotMinecraft) {
        if (packet.getColumn().getBiomeData() == null)
            return; //Not a full chunk.

        ccBotMinecraft.getPlayer().getChunkCache().addChunk(packet.getColumn());
    }
}

File: src/main/kotlin/com/andrew121410/mc/ccminecraftbot/packets/handle/OnServerChunkDataPacket.kt
package com.andrew121410.mc.ccminecraftbot.packets.handle

import com.andrew121410.mc.ccminecraftbot.CCBotMinecraft
import com.andrew121410.mc.ccminecraftbot.packets.PacketHandler
import com.github.steveice10.mc.protocol.packet.ingame.server.world.ServerChunkDataPacket

class OnServerChunkDataPacket : PacketHandler<ServerChunkDataPacket>() {
    override fun handle(packet: ServerChunkDataPacket, ccBotMinecraft: CCBotMinecraft) {
        if (packet.column.biomeData == null) return  //Not a full chunk.
        ccBotMinecraft.player.chunkCache.addChunk(packet.column)
    }
}